backend/database/wrapper/stats/player_stat_wrapper.py

```diff
@@ -176,41 +177,61 @@ def _create_group_stats_from_query(self, session, query, player_filter=None, rep
+        return stats.build_query(session).filter(PlayerGame.time_in_game > 0).count()
+
+    def _split_by_category(self, stats):
+        stats_dict = {}
+        for chart_metadata in player_group_stats_metadata:
+            if chart_metadata.stat_name not in stats:
+                continue
+            stats_dict[chart_metadata.stat_name] = {
+                "value": stats[chart_metadata.stat_name],
+                "subcategory": chart_metadata.subcategory
+            }
+        for stat in set(stats.keys()) - set([stat.stat_name for stat in player_group_stats_metadata]):
+            stats_dict[stat] = {
+                "value": stats[stat],
+                "subcategory": "Misc"
+            }
+        return stats_dict
+
     def _create_group_stats(self, session, player_filter=None, replay_ids=None):
         average = self._create_group_stats_from_query(session, self.get_player_stat_query(), player_filter, replay_ids)
         individual = self._create_group_stats_from_query(session, self.player_stats.individual_query, player_filter,
                                                          replay_ids)
-
         total = {}
         per_game = {}
         per_norm_game = {}
         per_minute = {}
 
-        factor_per_game = len(replay_ids) if replay_ids is not None else None
+        factor_per_game = self._get_counts(session, player_filter, replay_ids)
         factor_per_minute = individual['time_in_game'] / 60.0 if 'time_in_game' in individual else None
         factor_per_norm_game = factor_per_minute / 5 if factor_per_minute is not None else None
-
         for stat in self.get_player_stat_list():
-            inserted = False
             stat_query_key = stat.get_query_key()
             if stat_query_key not in individual or stat_query_key not in average:
                 continue
             if stat_query_key in self.replay_group_stats.grouped_stat_total:
                 if stat.is_percent or stat.is_averaged:
                     total[stat_query_key] = average[stat_query_key]
-                    inserted = True
                 else:
                     total[stat_query_key] = individual[stat_query_key]
-                    inserted = True
             if stat_query_key in self.replay_group_stats.grouped_stat_per_game and factor_per_game is not None:
                 per_game[stat_query_key] = individual[stat_query_key] / factor_per_game
-                inserted = True
             if stat_query_key in self.replay_group_stats.grouped_stat_per_minute and factor_per_minute is not None:
                 per_minute[stat_query_key] = individual[stat_query_key] / factor_per_minute
-                inserted = True
-            if not inserted and factor_per_norm_game is not None:
-                per_norm_game[stat_query_key] = individual[stat_query_key] / factor_per_norm_game
-
+            if factor_per_norm_game is not None:
+                if stat_query_key in self.replay_group_stats.grouped_stat_total and (
```

Comment:
the way this is formatted is very confusing.
Can you maybe reformat the entire thing so that we don't end a line with a (

If you wrap the entire statement in a () then we dont end the line with a \ either

Reply:
blame pycharm for the formatting